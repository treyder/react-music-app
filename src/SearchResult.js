import React from 'react';
import MovieItemList from './MovieItemList';

class SearchResult extends React.Component {

    constructor(props) {
        super(props);
        this.state = {
            sortBy: "date"
        }
    }

    render() {
        if (this.props.moviesList == undefined) {
            return (
                <div className="Search-result">
                    Start searching ...
                </div>
            );
        } else if (this.props.moviesList.length == 0) { 
            return (
                <div className="Search-result">
                    No films found
                </div>
            );
        }
        else {
            return (
                <div id="searchResultList" className="Search-result SearchResult-column">
                    {this.props.moviesList.map((item) => <MovieItemList key={item.id} title={item.title} img={item.poster_path} genre={item.genres} rating={item.vote_count} />)}
                </div>
            );
        }
    }
}

export default SearchResult;