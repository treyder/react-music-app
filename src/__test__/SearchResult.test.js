import React from "react";
import renderer from 'react-test-renderer';
import SearchResult from "../SearchResult";

describe("SearchResult", () => {
    
    it("renders initial state to - Start searching", () => {
        const tree = renderer.create(
            <SearchResult />
        ).toJSON();
        expect(tree).toMatchSnapshot();
    });

    it("renders in case of no results - No films found", () => {
        const tree = renderer.create(
            <SearchResult moviesList={[]}/>
        ).toJSON();
        expect(tree).toMatchSnapshot();
    });

    it("renders list of movies - List", () => {
        const tree = renderer.create(
            <SearchResult moviesList={[{
                id:0,
                title: "some title",
                poster_path: "img_src",
                genres: ["test genre"],
                vote_count: 1055
            },
            {
                id:1,
                title: "some title2",
                poster_path: "img_src2",
                genres: ["test genre2"],
                vote_count: 1056
            }
        ]}/>
        ).toJSON();
        expect(tree).toMatchSnapshot();
    });

});