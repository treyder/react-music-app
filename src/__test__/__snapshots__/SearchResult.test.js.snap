// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchResult renders in case of no results - No films found 1`] = `
<div
  className="Search-result"
>
  No films found
</div>
`;

exports[`SearchResult renders initial state to - Start searching 1`] = `
<div
  className="Search-result"
>
  Start searching ...
</div>
`;

exports[`SearchResult renders list of movies - List 1`] = `
<div
  className="Search-result SearchResult-column"
  id="searchResultList"
>
  <div
    className="movieItemList"
  >
    <img
      src="img_src"
    />
    some title
     
    1055
    test genre
  </div>
  <div
    className="movieItemList"
  >
    <img
      src="img_src2"
    />
    some title2
     
    1056
    test genre2
  </div>
</div>
`;
